src/index.jsx

```diff
@@ -69,22 +56,41 @@ function configure() {
   };
 }
 
-apiClient.ensurePublicOrAuthenticationAndCookies(
-  window.location.pathname,
-  (accessToken) => {
-    const { store, history } = configure();
+function initialize(accessToken) {
+  const { store, history } = configure();
+
+  if (accessToken) {
+    ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
+    identifyAuthenticatedUser(accessToken.userId);
+    sendPageEvent();
+  } else {
+    // This should never happen, but it does sometimes.
+    // Add logging to learn more.
+    throw new Error('Empty accessToken returned from ensurePublicOrAuthenticationAndCookies callback.');
+  }
+}
+
+try {
+  if (configuration.ENVIRONMENT !== 'production') {
+    import(/* webpackChunkName: "react-axe" */ 'react-axe').then(({ default: axe }) =>
+      axe(React, ReactDOM, 1000));
+  }
 
-    if (accessToken) {
-      ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
-      identifyAuthenticatedUser(accessToken.userId);
-      sendPageEvent();
-    } else {
-      // This should never happen, but it does sometimes.
-      // Add logging to learn more.
-      ReactDOM.render(<ErrorPage />, document.getElementById('root'));
-      NewRelicLoggingService.logError('Empty accessToken returned from ' +
-        'ensurePublicOrAuthenticationAndCookies callback.');
-      identifyAnonymousUser();
-    }
-  },
-);
+  apiClient = getAuthenticatedAPIClient({
+    appBaseUrl: configuration.BASE_URL,
+    authBaseUrl: configuration.LMS_BASE_URL,
+    loginUrl: configuration.LOGIN_URL,
+    logoutUrl: configuration.LOGOUT_URL,
+    csrfTokenApiPath: configuration.CSRF_TOKEN_API_PATH,
+    refreshAccessTokenEndpoint: configuration.REFRESH_ACCESS_TOKEN_ENDPOINT,
+    accessTokenCookieName: configuration.ACCESS_TOKEN_COOKIE_NAME,
+    userInfoCookieName: configuration.USER_INFO_COOKIE_NAME,
+    csrfCookieName: configuration.CSRF_COOKIE_NAME,
+    loggingService: NewRelicLoggingService,
+  });
+
+  apiClient.ensurePublicOrAuthenticationAndCookies(window.location.pathname, initialize);
+} catch (e) {
+  ReactDOM.render(<ErrorPage />, document.getElementById('root'));
+  NewRelicLoggingService.logError(e.message);
```

Comment:
Note that we're using NewRelicLoggingService directly here (which I didn't change).  In the long term, we might need to think about how we use frontend-logging's interface, rather than calling the NewRelic service directly.

This may be complicated by the fact that the logging service needs to be configured _in_ the code we're wrapping with this try/catch.